### ISSUE TITLE
feat: add names to all events and functions signatures inputs

### ISSUE THREAD

**Issue by sammccord** [boostxyz/boost-protocol#191]
### Description

Frontend needs to show detailed info about action steps. They have the signature, and `@boostxyz/signatures` to get the event/function name, but would need names for individual inputs to both help users create boosts, and our devs to build richer ui.	

### Suggested Solution

Add names to all events and functions in `boost-protocol/packages/signatures/manifests`

ie
`Transfer(address indexed,address indexed,uint256 indexed)`
becomes
`Transfer(address indexed from,address indexed to,uint256 indexed amount)`

**Comment by Quazia** [comment]
This will cause `args` to get returned as K:V pairs instead of an array - they're unnamed for internal use intentionally. We could maintain both or maintain the named list and employ a helper to strip out the names. This is related to the signatures ergonomics work @topocount was going to focus on next. The ask here is basically that we need to serve the event as a human readable ABI _with_ params in some way right?

**Comment by Quazia** [comment]
Another somewhat related thing I noticed recently too is there are some issues if we have the same event selector but the configuration of what is indexed is different - i.e. `Transfer(address indexed,address indexed,uint256 indexed)` vs `Transfer(address indexed,address indexed,uint256)`

**Comment by topocount** [comment]
@Quazia We have a ticket for that isssue with indexed vs nonindexed args here: https://linear.app/rh-app/issue/BOOST-4764/modify-the-eventaction-validators-to-account-for-indexed-args-through

**Comment by sammccord** [comment]
> This will cause `args` to get returned as K:V pairs instead of an array - they're unnamed for internal use intentionally. We could maintain both or maintain the named list and employ a helper to strip out the names. This is related to the signatures ergonomics work @topocount was going to focus on next. The ask here is basically that we need to serve the event as a human readable ABI _with_ params in some way right?

Yeah, Boost detail designs, and create flow rely on input names to label actions to be less error-prone and make the ui richer.

**Comment by topocount** [comment]
@sammccord this needs to be downstream of the work on signature package eronomics, as it'll break EventAction right now.

**Comment by soulstar9792** [comment]
May I ask if the new-comers work for a paid position?

**Comment by sammccord** [comment]
> May I ask if the new-comers work for a paid position?

@soulstar9792 Thanks for inquiring. We are not currently hiring freelance contributors without prior history of contributions to the repository.

**Comment by Antrikshgwal** [comment]
Hey @sammccord !!
Is this issue still open? If yes, can you please assign me?
